ui/packages/shared/profile/src/ProfileMetricsGraph/index.tsx

```diff
@@ -42,36 +42,39 @@ export const useQueryRange = (
   queryExpression: string,
   start: number,
   end: number
-): IQueryRangeResult => {
-  const [isLoading, setIsLoading] = useState<boolean>(false);
-  const [error, setError] = useState<any>(null);
-  const [response, setResponse] = useState<QueryRangeResponse | null>(null);
+): IQueryRangeState => {
+  const [state, setState] = useState<IQueryRangeState>({
+    response: null,
+    isLoading: true,
+    error: null,
+  });
```

Comment:
not just cleaner, before we *never* cleared the error state

Reply:
yes yes, great catch! By "cleaner" I meant exactly that! Not just how the code looks, but more-so how it works - saving loading/response/error in a single "state" rather than separate because we can't have two of these simultaneously. If we are loading, we want to clear the previous results/error, if we have a result or error, we are not loading, so and so forth. Definitely helps us avoid logical errors like the one you mentioned of never clearing the error state